internal/plan/plan.go

```diff
@@ -172,21 +205,21 @@ func CombineLayers(layers ...*Layer) (*Layer, error) {
 // An error is returned when a provided service name does not exist, or there
 // is an order cycle involving the provided service or its dependencies.
 func (p *Plan) StartOrder(names []string) ([]string, error) {
-	return order(p.Services, names, false)
+	return order(p.Services, names, false, true)
 }
 
 // StopOrder returns the required services that must be stopped for the named
 // services to be properly stopped, in the order that they must be stopped.
 // An error is returned when a provided service name does not exist, or there
 // is an order cycle involving the provided service or its dependencies.
-func (p *Plan) StopOrder(names []string) ([]string, error) {
-	return order(p.Services, names, true)
+func (p *Plan) StopOrder(names []string, includeDependants bool) ([]string, error) {
+	return order(p.Services, names, true, includeDependants)
 }
 
-func order(services map[string]*Service, names []string, stop bool) ([]string, error) {
+func order(services map[string]*Service, names []string, stop bool, includeRelated bool) ([]string, error) {
 	// For stop, create a list of reversed dependencies.
 	predecessors := map[string][]string(nil)
-	if stop {
+	if stop && includeRelated {
```

Comment:
While reviewing the follow up PR, I started to suspect that this logic can't be right, as it's blocking ordering knowledge from the algorithm altogether.  We need to feed the hierarchy into the dependency algorithm even if we're not going to be using all of it, because we don't know which paths are present in the graph. In other words, even if we only care about A and D, we need to have A, B, C, and D into the dependency logic, so that we can find the path betweem A and D and order them.  As such, I suggest leaving StartOrder and StopOrder alone, and just having a helper function that when given two lists L1 and L2, will just reorder L1 according to the order of these elements L2.